cmd/osquery-perf/agent.go

```diff
@@ -459,6 +527,26 @@ func (a *agent) runOrbitLoop() {
 	}
 }
 
+func (a *agent) runMDMLoop() {
+	mdmCheckInTicker := time.Tick(a.MDMCheckInInterval)
+
+	var mdmCommandPayload *mdm.CommandPayload
+	for range mdmCheckInTicker {
+		var err error
+		if mdmCommandPayload == nil {
+			mdmCommandPayload, err = a.mdmDevice.Idle()
+		} else {
+			// This MDM device acknowledges every command.
+			a.stats.IncrementMDMCommandsReceived()
+			mdmCommandPayload, err = a.mdmDevice.Acknowledge(mdmCommandPayload.CommandUUID)
+		}
+		if err != nil {
+			log.Printf("MDM request failed: %s\n", err)
+			a.stats.IncrementMDMErrors()
+		}
+	}
+}
```

Comment:
we already chatted about this block of code, but I'm realizing this now (sorry)

this can be considered a nit: in reality, once the device acknowledges a command, unless it looses the network connection, it generally keeps pulling commands until the server stops sending more commands.

so the `for range mdmCheckInTicker` can be used to send an `Idle` from time to time, but once you get a command we could do:

```
for mdmCommandPayload != nil {
		a.stats.IncrementMDMCommandsReceived()
		mdmCommandPayload, err = a.mdmDevice.Acknowledge(mdmCommandPayload.CommandUUID)
}
```

I think that whether or not this is a big deal for the load test depends on `a.MDMCheckInInterval`

Reply:
Good idea! Pushed changes: [f4cc406](https://github.com/fleetdm/fleet/pull/11672/commits/f4cc406a3846ad834e9d6b728af704aa09ce0739).